codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/waiters/KotlinJmespathExpressionVisitor.kt

```diff
@@ -286,6 +286,44 @@ class KotlinJmespathExpressionVisitor(
             VisitedExpression(addTempVar("merge", objects.mergeProperties()), isObject = true)
         }
 
+        "max" -> {
+            writer.addImport(RuntimeTypes.Core.Utils.max)
+            val list = expression.singleArg()
+            list.dotFunction(expression, "max()")
+        }
+
+        "min" -> {
+            writer.addImport(RuntimeTypes.Core.Utils.min)
+            val list = expression.singleArg()
+            list.dotFunction(expression, "min()")
+        }
+
+        "reverse" -> {
+            val listOrString = expression.singleArg()
+            listOrString.dotFunction(expression, "reversed()")
+        }
```

Comment:
**Question**: Do these not need null guards?

Reply:
The `dotFunction` function takes care of that:

    private fun VisitedExpression.dotFunction(expression: FunctionExpression, expr: String, elvisExpr: String? = null, isObject: Boolean = false): VisitedExpression {
        val dotFunctionExpr = ensureNullGuard(shape, expr, elvisExpr)
         ...
    }